doc/developers_gocean1p0_stencils.rst

```diff
@@ -0,0 +1,44 @@
+*** Please add this documentation to the GOcean section of the
+developers documentation when the developers documentation is added to
+master. ***
+
+The logic for checking and storing stencil information is contained in
+the `GOStencil` class in the `gocean1p0.py` file. This class is
+written so that it can be instantiated in isolation from the rest of
+`PSyclone` with the actual input data being performed by a `load()`
+method. This should help with testing and for adding new types of data
+loading in the future if required.
+
+The strategy for capturing stencil information differs between the
+`GOcean1.0` and `Dynamo0.3` APIs. The `GOcean` strategy of keeping
+everything in a stencil class is arguably better.
+
+An instance of the `GOStencil` class is attached to all field and
+scalar argument objects (the `GOKernelArgument` class) with the name
+`stencil`.
+
+A grid argument (class `GOKernelGridArgument`) will not have a stencil
+attribute, however a scalar and a field argument will always have a
+stencil attribute with a valid `GOStencil` object. In this case the
+stencil object can be queried to see if the argument has a stencil or
+not using the `has_stencil` method. This might seem strange, but it
+was the natural way to implement stencil support given the use of
+`pointwise` for a stencil attribute (which means that there is no
+stencil access).
+
+The use of `pointwise` is redundant for scalars and also for fields
+with `write` or `readwrite` access. At the moment this information
```

Comment:
perhaps: "... access (because kernels are only permitted to write to the 'current' i,j'th grid-point in the field)."